In [1]:
import cv2
import numpy as np

img = cv2.imread('image.jpg')

# get grayscale image
def get_grayscale(image):
    return cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)

# noise removal
def remove_noise(image):
    return cv2.medianBlur(image,5)
 
#thresholding
def thresholding(image):
    return cv2.threshold(image, 0, 255, cv2.THRESH_BINARY + cv2.THRESH_OTSU)[1]

#dilation
def dilate(image):
    kernel = np.ones((5,5),np.uint8)
    return cv2.dilate(image, kernel, iterations = 1)
    
#erosion
def erode(image):
    kernel = np.ones((5,5),np.uint8)
    return cv2.erode(image, kernel, iterations = 1)

#opening - erosion followed by dilation
def opening(image):
    kernel = np.ones((5,5),np.uint8)
    return cv2.morphologyEx(image, cv2.MORPH_OPEN, kernel)

#canny edge detection
def canny(image):
    return cv2.Canny(image, 100, 200)

#skew correction
def deskew(image):
    coords = np.column_stack(np.where(image > 0))
    angle = cv2.minAreaRect(coords)[-1]
    if angle < -45:
        angle = -(90 + angle)
    else:
        angle = -angle
    (h, w) = image.shape[:2]
    center = (w // 2, h // 2)
    M = cv2.getRotationMatrix2D(center, angle, 1.0)
    rotated = cv2.warpAffine(image, M, (w, h), flags=cv2.INTER_CUBIC, borderMode=cv2.BORDER_REPLICATE)
    return rotated

#template matching
def match_template(image, template):
    return cv2.matchTemplate(image, template, cv2.TM_CCOEFF_NORMED) 

In [2]:
import cv2 
import pytesseract

img = cv2.imread('image.jpg')

# Adding custom options
custom_config = r'--oem 3 --psm 6'
text=pytesseract.image_to_string(img, config=custom_config)
print(text)

iSe realiz6 tu pago con éxito!
Tunimero de operation es #00040193. Ademés so ha erviado una constenca al
correo electrcrico: peat*+*™*"*@qmall.com, Recuorda que ol monto total a
cambio os el importe quo se carga tu cuenta afoctando ol tipo de cambio.
& Origen
cuta de orgen
Ahorro Soles 215-07081220-0-44
a Destino
empress categoa
Universidad Nacional De San Agustin Universidades
‘Unsa - Virtual CAHUANA/NINA, JASON PAUL
cécigo de uns
cazimoog
{9 Documento(s) pagados
cezmoosrension 5/4600 5/000 s/000 5/000 sofu/z0 5) 48000
aoc is Monto total pagadoS/ 920.00



In [3]:
image = cv2.imread('image.jpg')

gray = get_grayscale(image)
thresh = thresholding(gray)
opening = opening(gray)
canny = canny(gray)

'iSe realiz6 tu pago con éxito!\nTunimero de operation es #00040193. Ademés so ha erviado una constenca al\ncorreo electrcrico: peat*+*™*"*@qmall.com, Recuorda que ol monto total a\ncambio os el importe quo se carga tu cuenta afoctando ol tipo de cambio.\n& Origen\ncuta de orgen\nAhorro Soles 215-07081220-0-44\na Destino\nempress categoa\nUniversidad Nacional De San Agustin Universidades\n‘Unsa - Virtual CAHUANA/NINA, JASON PAUL\ncécigo de uns\ncazimoog\n{9 Documento(s) pagados\ncezmoosrension 5/4600 5/000 s/000 5/000 sofu/z0 5) 48000\naoc is Monto total pagadoS/ 920.00\n\x0c'

In [4]:
cv2.imshow('img', gray)
cv2.waitKey(0)

text=pytesseract.image_to_string(gray, config=custom_config)
print(text)

Qt: Session management error: Could not open network socket


iSe realiz6 tu pago con éxito!
Tu nimero de operaciones £00040193. Ademés so ha enviado una canstenca al
correo electrcrico: peat*+*™***@gmall.com, Recuords que ol monto total a
cambio os el importe quo se carga tu cuenta afoctando ol tipo de camo.
© Origen
cunts de organ
Ahorro Soles 215-07081220-0-44
Destino
empresa categoa
Universidad Nacional De San Agustin Universidades
Unsa - Virtual CAHUANA/NINA, JASON PAUL
cécigo de uss
cezimoog
{9 Documento(s) pagados
cexmousensione s/ae000 s/00 s/o 5/000 son/zoc! 5) 48000
cezmoosrension 5/4600 5/000 s/000 5/000 sofu/z0 5) 48000
en sia datatiacia ia oa Monto total pagadoS/ 920.00



In [5]:
cv2.imshow('img', thresh)
cv2.waitKey(0)

text=pytesseract.image_to_string(thresh, config=custom_config)
print(text)

iSe realizé tu pago con éxito!
Tu nomero de operacon es ROODAOTSD, Ademds sha erviado una canstanca al
correo slertranica: part**"-*@qmall.com, Recuorda que ol manto total a
cambio es el importa qua se carga tu cuenta afectando ol tia de camtio.
& Origen
cute de organ
Ahorro Soles 215-07081220-0-44
“@ Destino
empresa cenegora
Universidad Nacional Oe Son Agustin Universidades
‘Unsa - Virtual CAHUANA/NINA, JASON PAUL
cétigo de uss
cezimong
(J Documento(s) pagados
cexmovseensions —s/asn0 s/o s/on0 5/000 so/nvanc 3) as000
cerwovsreNion7 —s/as000 | s/aoo s/on0 s/000 sahz/zoa1 5) as000
11 enviar sonstantia de soe Monte total pagadoS/ 920.00



In [6]:
cv2.imshow('img', opening)
cv2.waitKey(0)

text=pytesseract.image_to_string(opening, config=custom_config)
print(text)

e—___e——_®
ie remia ts pepe com dentin!
Tene Se A A oe ee
ee
eee eS a i
= Onigen
ee
ae es LL
~w Destin
— oe
Cat Mines ae feat yen a a
a iii
om Pa
<i
ee
() Decent) pages
jee eee re eo eee
ee ee
2 ee Mere eel) SOOO
==) GED



In [7]:
cv2.imshow('img', canny)
cv2.waitKey(0)

text=pytesseract.image_to_string(canny, config=custom_config)
print(text)

aa ox exes
a YY ae
‘Oe
ASS en Rac cee ad
Rea oop raven os cd Oc eae oan ee ny a eomec Le nee cy
oo Na ol eee ee RL Ls ox)
Pica seater ra vowel con moana ogee na aca no
f=Folngen)
Cont)
eer ea Cay
Emr
coe cy
Doce ccc oe De
rood Sed
Cee Ce DN CCR
Reece od
Ce)
Penne eared
od a
Co CR RL)
Ce CL ee L)
See eae) esata) BTN)
owe



In [8]:
#Getting boxes around text
#this process is to see where is working our tesseract
import cv2
import pytesseract

img = cv2.imread('image.jpg')

h, w, c = img.shape
boxes = pytesseract.image_to_boxes(img) 
for b in boxes.splitlines():
    b = b.split(' ')
    img = cv2.rectangle(img, (int(b[1]), h - int(b[2])), (int(b[3]), h - int(b[4])), (0, 255, 0), 2)

cv2.imshow('img', img)
cv2.waitKey(0)

48

In [1]:
import cv2
import pytesseract
from pytesseract import Output

img = cv2.imread('image.jpg')

d = pytesseract.image_to_data(img, output_type=Output.DICT)
print(d.keys())

dict_keys(['level', 'page_num', 'block_num', 'par_num', 'line_num', 'word_num', 'left', 'top', 'width', 'height', 'conf', 'text'])


In [2]:
n_boxes = len(d['left'])
for i in range(n_boxes):
    if int(d['conf'][i]) > 60:
        (x, y, w, h) = (d['left'][i], d['top'][i], d['width'][i], d['height'][i])
        img = cv2.rectangle(img, (x, y), (x + w, y + h), (0, 255, 0), 2)

cv2.imshow('img', img)
cv2.waitKey(0)

Qt: Session management error: Could not open network socket


48

In [ ]:
install: 
tesseract
pytesseract